### PR TITLE
Fluid container error events [main -> lts]

### PR DESCRIPTION
**Pull request by ssimic2**
Bringing https://github.com/microsoft/FluidFramework/pull/12256 and https://github.com/microsoft/FluidFramework/pull/11554 to LTS

### CODE REVIEW

File: packages/framework/fluid-framework/src/containerDefinitions.ts [hunk 1]
```diff
@@ -3,4 +3,8 @@
  * Licensed under the MIT License.
  */
 
-export { AttachState } from "@fluidframework/container-definitions";
+export {
+    AttachState,
```
Comment (ssimic2) [inline packages/framework/fluid-framework/src/containerDefinitions.ts:7]: This should be unified * exports, but it looks like we have some conflicts across container-definitions / driver-definitions.

Comment (tylerbutler) [reply packages/framework/fluid-framework/src/containerDefinitions.ts:7]: We're switching to explicit exports across the repo anyway.

File: packages/framework/fluid-framework/src/index.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@
  */
 
 export * from "./containerDefinitions";
+export * from "./driverDefinitions";
 export * from "./containerLoader";
 export * from "./fluidStatic";
```
Comment (tylerbutler) [inline packages/framework/fluid-framework/src/index.ts:17]: nit

```suggestion
export * from "./containerLoader";
export * from "./driverDefinitions";
export * from "./fluidStatic";
```

File: packages/framework/fluid-static/src/fluidContainer.ts [hunk 3]
```diff
@@ -5,7 +5,12 @@
 import { TypedEventEmitter } from "@fluidframework/common-utils";
 import { IFluidLoadable } from "@fluidframework/core-interfaces";
 import { IEvent, IEventProvider } from "@fluidframework/common-definitions";
-import { AttachState, IContainer, ConnectionState } from "@fluidframework/container-definitions";
+import {
+    AttachState,
+    IContainer,
+    ICriticalContainerError,
+    ConnectionState,
```
Comment (tylerbutler) [inline packages/framework/fluid-static/src/fluidContainer.ts:12]: nit

```suggestion
    AttachState,
    ConnectionState,
    IContainer,
    ICriticalContainerError,
```

**Review by tylerbutler — APPROVED** [review]
total nits about alphabetization only, and since this is a cherry-pick, maybe just ignore them.

File: packages/framework/fluid-framework/src/containerDefinitions.ts (hunk 1 above)
Comment (ssimic2) [reply packages/framework/fluid-framework/src/containerDefinitions.ts:7]: The fluid-framework is somewhat specific is all that it does is re-exports. We should reevaluate if this is the best way forward.

**Pull request merged**